### PR TITLE
Modified backend transform() to accept output arg

### PR DESCRIPTION
**Pull request by mlloreda**

### CODE REVIEW

File: src/api/c/transform.cpp [hunk 1]
```diff
@@ -19,10 +19,10 @@ using af::dim4;
 using namespace detail;
 
 template<typename T>
-static inline af_array transform(const af_array in, const af_array tf, const af::dim4 &odims,
+static inline void transform(af_array *out, const af_array in, const af_array tf, const dim4 &odims,
                                  const af_interp_type method, const bool inverse, const bool perspective)
```
Comment (FloopCZ) [inline src/api/c/transform.cpp]:
```suggestion
                              const af_interp_type method, const bool inverse, const bool perspective)
```

File: src/backend/opencl/transform.cpp [hunk 2]
```diff
@@ -38,12 +36,11 @@ namespace opencl
...
             AF_ERROR("Unsupported interpolation type", AF_ERR_ARG);
         }
-        return out;
     }
 
 
 #define INSTANTIATE(T)                                                                  \
-    template Array<T> transform(const Array<T> &in, const Array<float> &tf,             \
+    template void transform(Array<T> &out, const Array<T> &in, const Array<float> &tf,  \
                                 const af::dim4 &odims, const af_interp_type method,     \
```
Comment (FloopCZ) [inline src/backend/opencl/transform.cpp]:
```suggestion
                             const af::dim4 &odims, const af_interp_type method,     \
```

File: src/backend/opencl/transform.cpp [hunk 3]
```diff
@@ -38,12 +36,11 @@ namespace opencl
...
                                 const bool inverse, const bool perspective);
```
Comment (FloopCZ) [inline src/backend/opencl/transform.cpp]:
```suggestion
                             const bool inverse, const bool perspective);
```

**Comment by mlloreda** [comment]
Thanks for feedback, @FloopCZ! Addressed.

**Comment by umar456** [comment]
Could you add tests to verify that these changes worked? You should look at the approx1 PR and look at the type of tests there. Ideally we want to make these tests as low costs as possible so it would be nice if you and @mark-poscablo can discuss making tests more succinct.

**Pull request merged**